frappe/website/doctype/website_theme/website_theme.py

```diff
@@ -11,7 +11,7 @@ class WebsiteTheme(Document):
 	def validate(self):
 		self.validate_if_customizable()
 		self.render_theme()
-		self.validate_theme()
+		self.generate_bootstrap_theme()
```

Comment:
Why generate theme on every save? Shouldn't we check if something changed or not before generating the theme?

Reply:
The current approach checks only for changes from the user side, but the included scss from the backend might change as well. If we keep the check, the user is forced to change his scss to get the changes introduced in the included stylesheets by, say, an update.

Reply:
Yes, this doesn't take into account the changes in the core website.scss. But this doesn't solve that problem either, this is dependent on the user to go to the Website Theme and click Save.

Maybe we could do an after migrate hook which will check if the website.scss has changed and will regenerate the theme files?

Reply:
Right, I'll have a look at it.